### ISSUE TITLE
twisted.internet.stdio should be ported to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7899]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7899|
|Type|enhancement|
|Created|2015-05-17 10:30:57Z|
|Branch|https://github.com/twisted/twisted/tree/stdio-py3-7899|

<details><summary>Searchable metadata</summary>

```
trac-id__7899 7899
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_stdio_py3_7899 branches/stdio-py3-7899
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1431858657314573 1431858657314573
changetime__1431950042992913 1431950042992913
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [44842]) Branching to stdio-py3-7899.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Changes made, builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)<a name="note_3"></a>|@moshez set owner to @hawkowl|
|-|-|

Hi HawkOwl, thanks for working on this!

I think this can go in as is.

One thing you might want to improve:

1. Instead of self.addCleanup(self.extraFile.close), maybe define an inline function and close + remove?

Since anyway your changes are only making the tests better, I don't think this should be a show-stopper. Please either merge as-is, or change above and merge.

Moshe Z.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Hi moshez, thanks for the review.

As it's using mktemp, it will be removed automatically. :)

Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44851]) Merge stdio-py3-7899: Port twisted.internet.stdio to Python 3

Author: hawkowl
Reviewer: moshez
Fixes: [#7899](https://github.com/twisted/twisted/issues/7899)
